src/da/activate.ts

```diff
@@ -0,0 +1,126 @@
+      _folder: vscode.WorkspaceFolder | undefined,
+    ): vscode.ProviderResult<vscode.DebugConfiguration[]> {
+      return [
+        {
+          name: "Launch Rego Workspace",
+          request: "launch",
+          type: "opa-debug",
+          command: "eval",
+          query: "data",
+          bundlePaths: ["${workspaceFolder}"],
+          enablePrint: true,
+        },
+      ];
+    },
+  }, vscode.DebugConfigurationProviderTriggerKind.Dynamic));
+
+  context.subscriptions.push(vscode.debug.registerDebugConfigurationProvider("opa-debug", {
+    provideDebugConfigurations(
+      _folder: vscode.WorkspaceFolder | undefined,
+    ): vscode.ProviderResult<vscode.DebugConfiguration[]> {
+      return [
+        {
+          name: "Launch Rego Workspace",
+          request: "launch",
+          type: "opa-debug",
+          command: "eval",
+          query: "data",
+          bundlePaths: ["${workspaceFolder}"],
+          inputPath: "${workspaceFolder}/input.json",
+          enablePrint: true,
+        },
+      ];
+    },
+  }, vscode.DebugConfigurationProviderTriggerKind.Initial));
+}
+
+class OpaDebugConfigurationProvider implements vscode.DebugConfigurationProvider {
+  /**
+   * Massage a debug configuration just before a debug session is being launched,
+   * e.g. add all missing attributes to the debug configuration.
+   */
+  resolveDebugConfiguration(
+    _folder: vscode.WorkspaceFolder | undefined,
+    config: vscode.DebugConfiguration,
+    _token?: vscode.CancellationToken,
+  ): vscode.ProviderResult<vscode.DebugConfiguration> {
+    // if launch.json is missing or empty
+    if (!config.type && !config.request && !config.name) {
+      const editor = vscode.window.activeTextEditor;
+      if (editor && editor.document.languageId === "rego") {
+        config.type = "opa-debug";
+        config.name = "Launch";
+        config.request = "launch";
+        config.command = "eval";
+        config.query = "data";
+        config.bundlePaths = ["${workspaceFolder}"];
+        config.stopOnEntry = true;
+        config.enablePrint = true;
+      }
+    }
+
+    if (config.request === "attach" && !config.program) {
+      return vscode.window.showInformationMessage("Cannot find a program to debug").then((_) => {
+        return undefined; // abort launch
+      });
+    }
+
+    if (config.request === "launch" && !config.dataPaths && !config.bundlePaths) {
+      return vscode.window.showInformationMessage("Cannot find Rego to debug").then((_) => {
+        return undefined; // abort launch
+      });
+    }
+
+    return config;
+  }
+}
+
+class OpaDebugAdapterExecutableFactory implements vscode.DebugAdapterDescriptorFactory {
+  createDebugAdapterDescriptor(
+    _session: vscode.DebugSession,
+    executable: vscode.DebugAdapterExecutable | undefined,
+  ): vscode.ProviderResult<vscode.DebugAdapterDescriptor> {
+    if (!executable) {
+      promptForUpdateRegal(minimumSupportedRegalVersion);
```

Comment:
Do we need to allow the user to configure another debug adapter, similar to the `opa.languageServers` option for language servers?

Reply:
If someone comes along with a different implementation, we can accomodate that then. For now, I'd say we leave it be.